compute/cloud-client/src/main/java/compute/disks/CreateSnapshot.java

```diff
@@ -0,0 +1,134 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package compute.disks;
+
+// [START compute_snapshot_create]
+
+import com.google.cloud.compute.v1.Disk;
+import com.google.cloud.compute.v1.DisksClient;
+import com.google.cloud.compute.v1.Operation;
+import com.google.cloud.compute.v1.RegionDisksClient;
+import com.google.cloud.compute.v1.Snapshot;
+import com.google.cloud.compute.v1.SnapshotsClient;
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class CreateSnapshot {
+
+  public static void main(String[] args)
+      throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    // TODO(developer): Replace these variables before running the sample.
+    // You need to pass `zone` or `region` parameter relevant to the disk you want to
+    // snapshot, but not both. Pass `zone` parameter for zonal disks and `region` for
+    // regional disks.
+
+    // Project ID or project number of the Cloud project you want to use.
+    String projectId = "YOUR_PROJECT_ID";
+
+    // Name of the disk you want to create.
+    String diskName = "YOUR_DISK_NAME";
+
+    // Name of the snapshot that you want to create.
+    String snapshotName = "YOUR_SNAPSHOT_NAME";
+
+    // The zone of the source disk from which you create the snapshot (for zonal disks).
+    String zone = "europe-central2-b";
+
+    // The region of the source disk from which you create the snapshot (for regional disks).
+    String region = "your-disk-region";
+
+    // The Cloud Storage multi-region or the Cloud Storage region where you
+    // want to store your snapshot.
+    // You can specify only one storage location. Available locations:
+    // https://cloud.google.com/storage/docs/locations#available-locations
+    String location = "europe-central2";
```

Comment:
Are the regions and locations the same? As in are they from the same list? if so can we use one variable for both here?

Reply:
They are different in that locations might have dual or multiple regions. The location list may also vary with time and need not always be the same as the region list. 
Hence, I would prefer to keep it as two separate variables, if that makes sense?

Reply:
Sounds good to me